Client/Frontend/Browser/HomePageHelper.swift

```diff
@@ -50,4 +50,27 @@ class HomePageHelper {
         tab.loadRequest(NSURLRequest(URL: url))
     }
 
+    func openHomePage(inTab tab: Tab, withNavigationController navigationController: UINavigationController?) {
+        if isHomePageAvailable {
+            openHomePage(tab)
+        } else {
+            setHomePage(toTab: tab, withNavigationController: navigationController)
+        }
+    }
+
+    func setHomePage(toTab tab: Tab, withNavigationController navigationController: UINavigationController?) {
+        let alertController = UIAlertController(
+            title: Strings.SetHomePageDialogTitle,
+            message: Strings.SetHomePageDialogMessage,
+            preferredStyle: UIAlertControllerStyle.Alert)
+        alertController.addAction(
+            UIAlertAction(title: Strings.SetHomePageDialogNo, style: .Cancel) { (action) in
+                // Do nothing.
+            })
+        alertController.addAction(
+            UIAlertAction(title: Strings.SetHomePageDialogYes, style: .Default) { (action) in
+                self.currentURL = tab.url
```

Comment:
I have a feeling that this is a retain cycle since the navigation controller will hold onto the UIAlertController but UIAlertController -> Action -> self (Helper)